static/examples/audio.html

```diff
@@ -0,0 +1,188 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <style>
+      video {
+        height: 320px;
+        width: 480px;
+      }
+      .mute {
+        cursor: pointer;
+      }
+    </style>
+    <script type="text/javascript">
+      async function start() {
+        // Poly from https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/getUserMedia
+        // Older browsers might not implement mediaDevices at all, so we set an empty object first
+        if (navigator.mediaDevices === undefined) {
+          navigator.mediaDevices = {};
+        }
+
+        // Some browsers partially implement mediaDevices. We can't just assign an object
+        // with getUserMedia as it would overwrite existing properties.
+        // Here, we will just add the getUserMedia property if it's missing.
+        if (navigator.mediaDevices.getUserMedia === undefined) {
+          navigator.mediaDevices.getUserMedia = function(constraints) {
+
+            // First get ahold of the legacy getUserMedia, if present
+            let getUserMedia = navigator.webkitGetUserMedia || navigator.mozGetUserMedia;
+
+            // Some browsers just don't implement it - return a rejected promise with an error
+            // to keep a consistent interface
+            if (!getUserMedia) {
+              return Promise.reject(new Error('getUserMedia is not implemented in this browser'));
+            }
+
+            // Otherwise, wrap the call to the old navigator.getUserMedia with a Promise
+            return new Promise(function(resolve, reject) {
+              getUserMedia.call(navigator, constraints, resolve, reject);
+            });
+          }
+        }
+
+        let audioCtx = new (window.AudioContext || window.webkitAudioContext)();
+        let voiceSelect = document.getElementById("voice");
+        let source, stream;
+
+        /* ______________       _______________
+        * | Event Target | <--- | AudioContext |
+        * |  ____________|      |______________|
+         */
+
+        let distortion = audioCtx.createWaveShaper();
+        let gainNode = audioCtx.createGain();
+        let biquadFilter = audioCtx.createBiquadFilter();
+        let delay = audioCtx.createDelay();
+
+        // distortion curve for the waveshaper, thanks to Kevin Ennis
+        // http://stackoverflow.com/questions/22312841/waveshaper-node-in-webaudio-how-to-emulate-distortion
+        // 
+        // it would be cool to change this into a vocoder / DSP for autotuning
+        // http://blogs.zynaptiq.com/bernsee/pitch-shifting-using-the-ft/
+        // see also https://github.com/k15z/pitch-perfect-auto-tune
+
+        function makeDistortionCurve(amount) {
+          let k = typeof amount === 'number' ? amount : 50,
+            n_samples = 44100,
+            curve = new Float32Array(n_samples),
+            deg = Math.PI / 180,
+            i = 0,
+            x;
```

Comment:
Rather than declaring `x` here as part of the big `let` (see above comment), declare it as `const` in the `for` loop below. for example:

```
     for ( ... ) {
       const  x = i * 2 / n_samples - 1;
     }
```